File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/1. Learn Bit Manipulation/python/04.Check for the power of 2.py
"""QUESTION:
Given an integer `n`, return True if it is a power of two, otherwise False.

Approach:
A power of two has exactly one set bit. Check that `n > 0` and `n & (n - 1)` equals 0.

Time Complexity: O(1)
Space Complexity: O(1)
"""

def is_power_of_two(n: int) -> bool:
    if n <= 0:
        return False
    return (n & (n - 1)) == 0

File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/1. Learn Bit Manipulation/javascript/04.Check for the power of 2.js
/**
QUESTION:
Given an integer `n`, return true if it is a power of two, otherwise false.

Approach:
A power of two has exactly one set bit. Check `n > 0` and `n & (n - 1)` equals 0.

Time Complexity: O(1)
Space Complexity: O(1)
*/

function isPowerOfTwo(n) {
    if (n <= 0) return false;
    return (n & (n - 1)) === 0;
}

module.exports = { isPowerOfTwo };
